pkg/apis/application/v1alpha1/types.go

```diff
@@ -264,6 +264,8 @@ type SyncOperation struct {
 	// Source overrides the source definition set in the application.
 	// This is typically set in a Rollback operation and nil during a Sync operation
 	Source *ApplicationSource `json:"source,omitempty" protobuf:"bytes,7,opt,name=source"`
+	// Overrides sync source with a local directory for development
+	LocalManifests []string `json:"localManifests,omitempty" protobuf:"bytes,8,rep,name=localManifests"`
```

Comment:
Field name I think should just be `Manifests`. The term "local" is aside the point that they were explicitly supplied.

Also. Go style guidelines should use the field name as the first word of the sentence describing the field.

Description should indicate that this is an optional field.